File: backend/app/models/booking_items.py
from ..extensions import db
from sqlalchemy import Numeric


class BookingItem(db.Model):
    __tablename__ = "booking_items"
    id = db.Column(db.Integer, primary_key=True)
    booking_id = db.Column(db.Integer, db.ForeignKey("bookings.id"), nullable=False)
    booking = db.relationship("Booking", backref="items")
    ticket_type_id = db.Column(
        db.Integer, db.ForeignKey("event_tickets.id"), nullable=False
    )
    ticket_type = db.relationship("EventTicket")
    quantity = db.Column(db.Integer, nullable=False)
    price_at_purchase = db.Column(Numeric(10, 2), nullable=False)


@property
def total_price(self):
    return self.quantity * self.price_at_purchase

File: backend/internal/models/booking_items.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type BookingItem struct {
	ID              uint           `gorm:"primaryKey" json:"id"`
	BookingID       uint           `gorm:"not null" json:"booking_id"`
	Booking         *Booking       `gorm:"foreignKey:BookingID" json:"booking,omitempty"`
	TicketTypeID    uint           `gorm:"not null" json:"ticket_type_id"`
	TicketType      *EventTicket   `gorm:"foreignKey:TicketTypeID" json:"ticket_type,omitempty"`
	Quantity        int            `gorm:"not null" json:"quantity"`
	PriceAtPurchase float64        `gorm:"type:numeric(10,2);not null" json:"price_at_purchase"`
	CreatedAt       time.Time      `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt       time.Time      `gorm:"autoUpdateTime" json:"updated_at"`
	DeletedAt       gorm.DeletedAt `gorm:"index" json:"-"`
}

func (bi *BookingItem) TotalPrice() float64 {
	return float64(bi.Quantity) * bi.PriceAtPurchase
}
